x509cert/x509.go

```diff
@@ -19,11 +19,13 @@ import (
 
 // GenCACert creates the CA certificate given signer.
 func GenCACert(config *crypki.CAConfig, signer crypto.Signer, hostname string, ips []net.IP, pka crypki.PublicKeyAlgorithm) ([]byte, error) {
-	const validityPeriod = uint64(730 * 24 * 3600) // 2 years
+	if config.ValidityPeriod == 0 {
+		config.ValidityPeriod = uint64(730 * 24 * 3600) // 2 years by default
+	}
```

Comment:
Could we create a function which would set all the default values for `crypki.CAConfig` and make sure that it is invoked right after every time a `crypki.CAConfig` instance is created? Typically the default values of a config should be set right after the config is created rather than right before the config is going to be used because a config is likely to be used more than once. The current solution works because the config is only used in one function, but I think that setting the default values early on still has advantages as it improves readability.